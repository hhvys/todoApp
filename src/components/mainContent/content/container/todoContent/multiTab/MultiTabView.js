import React from 'react';
import CheckBox from '../../../../../logos/CheckBox';
import '../../../../../logos/logos.css';
import Star from '../../../../../logos/Star';
import Starred from "../../../../../logos/Starred";
import CheckedCheckBox from "../../../../../logos/CheckedCheckBox";
import './multiTab.css';
import '../../../../../logos/logos.css';
import Message from './Message';

const renderList = (todos, tabId, {onStarClick, onCheckClick}) => (
	todos.map(todo => (
		<li className="todo pl-3 d-flex justify-content-between" key={todo.todoId}>
			<div className="d-flex align-items-center">
				{
					todo.completed ?
						<CheckedCheckBox onClick={() => onCheckClick(tabId, todo.todoId)}/> :
						<CheckBox onClick={() => onCheckClick(tabId, todo.todoId)}/>
				}
				<div className="todo-content txt-overflow">{todo.text}</div>
			</div>
			<div className="pr-3">
				{
					todo.star ?
						<Starred onClick={() => onStarClick(tabId, todo.todoId)}/> :
						<Star onClick={() => onStarClick(tabId, todo.todoId)}/>
				}
			</div>
		</li>
	))
);

const renderContent = ({tabs, onTabClick, ...props}) => {
	console.log(tabs.length);
	if(!tabs.length)
		return <Message
								message="No Search Result"
								className="not-found"	/>
	return tabs.map(tab => (
		<div key={tab.tabId}>
			<div className="multi-tab-button mt-3" onClick={() => onTabClick(tab.tabId)}>
				{tab.tabName}
			</div>
			<div className="multi-tab-content">
				<ul className="mt-2">
					{renderList(tab.todos, tab.tabId, props)}
				</ul>
			</div>

		</div>
	))
};

const MultiTabView = (props) => {
	return (
		<div>
			{renderContent(props)}
		</div>
	);
};

export default MultiTabView;