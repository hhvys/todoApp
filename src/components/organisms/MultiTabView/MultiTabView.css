.div__multi__tab__view {
    margin-left: 300px;
    margin-top: 70px;
    height: 100%;
}

.div__multi__tab__view.collapsed {
    margin-left: 62px;
}