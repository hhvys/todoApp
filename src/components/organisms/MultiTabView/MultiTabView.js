import React from 'react';
import Button from "../../atoms/Button/Button";
import styles from './MultiTabView.mod.scss';
import {CHECK_BOX, CHECKED_CHECK_BOX} from "../../atoms/logos/constants";
import VerticalTab from '../../molecules/VerticalTab/VerticalTab';
import {INBOX_ID, SORT_BY} from "../../../actions/actionTypes"
import {STARRED, STAR} from "../../atoms/logos/constants";
import InputWithLabel from '../../molecules/InputWithLabel/InputWithLabel';
import NotFound from '../../molecules/NotFound/NotFound';

class MultiTabView extends React.Component {


	renderVerticalTab = ({
												 todos,
												 tabId,
												 onFooterSymbolClick,
												 onHeaderSymbolClick,
												 sortBy,
											 }) => {

		return (
			todos
				.sort((a, b) => {
					switch (sortBy) {
						case SORT_BY.SORT_PRIORITY:
							if (a.star && !b.star)
								return -1;
							if (!a.star && b.star)
								return 1;
							return 0;

						case SORT_BY.SORT_CREATION:
							if (a.completedTime > b.completedTime)
								return -1;
							return 1;

						case SORT_BY.SORT_ALPHA:
							if (a.text < b.text)
								return -1;
							return 1;
					}

				})
				.map(todo => {

					return (
						<VerticalTab key={todo.todoId}
												 className={styles.todo}
												 headerSymbol={todo.completed ? CHECKED_CHECK_BOX : CHECK_BOX}
												 footerSymbol={todo.star ? STARRED : STAR}
												 headerClass={styles.logo}
												 footerClass={styles.logo}
												 mainContent={todo.text}
												 onFooterSymbolClick={() => onFooterSymbolClick(tabId, todo.todoId)}
												 onHeaderSymbolClick={() => onHeaderSymbolClick(tabId, todo.todoId)}
						/>
					);
				})
		);
	};

	renderTabs({
							 tabs,
							 onFooterSymbolClick,
							 onHeaderSymbolClick,
							 sortBy,
							 onButtonClick,
							 searchQuery
						 }) {
		tabs = searchQuery.length !== 0 ?
			tabs
				.map(tab => ({
					...tab,
					todos: tab
						.todos
						.filter(todo => !todo.completed && (new RegExp(searchQuery).test(todo.text)))
				})) :
			tabs
				.map(tab => ({
					...tab,
					todos: tab
						.starredTodos
						.filter(todo => !todo.completed)
				}));

		tabs = tabs.filter(tab => tab.todos.length);

		if (tabs.length === 0 && searchQuery.length === 0)
			onButtonClick(INBOX_ID);

		if (tabs.length === 0)
			return (<NotFound/>);

		return tabs.map(tab => (
			<div key={tab.tabId}>
				<Button text={tab.tabName.toLocaleUpperCase()}
								className={"mt-4 mb-2"}
								onClick={() => onButtonClick(tab.tabId)}/>
				{
					this.renderVerticalTab({
						todos: tab.todos,
						tabId: tab.tabId,
						onFooterSymbolClick,
						onHeaderSymbolClick,
						sortBy
					})
				}
			</div>
		))
	}

	render() {
		const {
			tabs,
			sortBy,
			onHeaderSymbolClick,
			onFooterSymbolClick,
			searchQuery,
			activeTab,
			onButtonClick,
			collapsed,
			onInputSubmit
		} = this.props;
		return (
			<div className={`${styles.multiTabView} ${collapsed ? styles.collapsed : ''} mr-3`}>
				{
					searchQuery.length === 0 ?
						<InputWithLabel
							label={"+"}
							placeholder={"Add a starred to-do in Inbox..."}
							style={{fontSize: 16}}
							onSubmit={(value) => onInputSubmit(value)}
						/> :
						null
				}
				{this.renderTabs({
					tabs,
					sortBy,
					onHeaderSymbolClick,
					onFooterSymbolClick,
					searchQuery,
					activeTab,
					onButtonClick
				})}
			</div>
		);
	}
}

export default MultiTabView;
