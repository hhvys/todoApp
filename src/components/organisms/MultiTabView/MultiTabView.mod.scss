.multiTabView {
    margin-left: 300px;
    margin-top: 70px;
    height: 100%;
}

.collapsed {
    margin-left: 62px;
}

.logo {
    margin: 10px;
}

.todo {
    margin-top: 2px;
    background-color: #F7F7F7;
    border-radius: 5px;
    height: 46px;
}
