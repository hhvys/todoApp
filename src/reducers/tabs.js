import {
	ADD_TAB,
	CHANGE_TAB_NAME,
	ADD_TODO,
	TOGGLE_TODO,
	DELETE_TAB,
	COPY_TAB,
	STAR_TOGGLE_TODO, ADD_STARRED_TODO, ACTIVE_TODO
} from '../actions/actionTypes';

const todos = (state = [], action) => {
	switch (action.type) {
		case ADD_TODO:
			return (
				[
					...state,
					{
						todoId: action.todoId,
						text: action.text,
						completed: false,
						star: false,
						completedTime: undefined,
						createdTime: new Date(),
						active: false
					}
				]);
		case ADD_STARRED_TODO:
			return (
				[
					...state,
					{
						todoId: action.todoId,
						text: action.text,
						completed: false,
						star: true,
						completedTime: undefined,
						createdTime: new Date(),
						active: false
					}
				]);
		case TOGGLE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						completed: !todo.completed,
						completedTime: new Date()
					} :
					todo
			));
		case STAR_TOGGLE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						star: !todo.star
					} :
					todo
			));
		case ACTIVE_TODO:
			return state.map(todo => (
				todo.todoId === action.todoId ?
					{
						...todo,
						active: true
					} :
					{
						...todo,
						active: false
					}
			));
		default:
			return state;
	}
};

const initialState = [
	{
		tabId: 0,
		tabName: 'Inbox',
		todos: []
	},
	{
		tabId: 1,
		tabName: 'Starred',
		todos: []
	}
];

function tabs(state = initialState, action) {
	switch (action.type) {
		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: []
				}
			];
		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						tabId: tab.tabId,
						tabName:
						action.tabName,
						todos: tab.todos
					} :
					tab
			));
		case COPY_TAB:
			let fromTab =
				state.find(tab => tab.tabId === action.fromId);
			return (
				[
					...state,
					{
						tabId: action.toId,
						tabName: fromTab.tabName + " Copy",
						todos: [...fromTab.todos]
					}
				]
			);
		case DELETE_TAB:
			return state.filter(tab =>
				tab.tabId !== action.tabId);
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
		case STAR_TOGGLE_TODO:
		case ACTIVE_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action)
					} : tab
			));
		default:
			return state;
	}
}

export default tabs;