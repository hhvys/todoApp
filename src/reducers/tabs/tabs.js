import {combineReducers} from 'redux';
import todo from './todo';
import todos from './todos';
import {
	ADD_TAB,
	CHANGE_TAB_NAME,
	ADD_TODO,
	TOGGLE_TODO,
	DELETE_TAB,
	COPY_TAB,
	STAR_TOGGLE_TODO,
	ADD_STARRED_TODO,
	ACTIVE_TODO,
	TOGGLE_VISIBILITY_FILTER,
	INBOX_ID,
	STARRED_ID
} from '../../actions/actionTypes';


const initialState = [
	{
		tabId: INBOX_ID,
		tabName: 'Inbox',
		todos: [],
		starredTodos: []
	},
	{
		tabId: STARRED_ID,
		tabName: 'Starred',
		todos: [],
		starredTodos: []
	}
];

function tabInfo(state = initialState, action) {
	switch (action.type) {
		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: [],
					showCompletedTodo: true,
					starredTodos: []
				}
			];
		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						tabId: tab.tabId,
						tabName: action.tabName,
					} :
					tab
			));
		case COPY_TAB:
			let fromTab =
				state.find(tab => tab.tabId === action.fromId);
			return (
				[
					...state,
					{
						tabId: action.toId,
						tabName: fromTab.tabName + " Copy",
						todos: [...fromTab.todos],
						starredTodos: [...fromTab.starredTodos]
					}
				]
			);
		case DELETE_TAB:
			return state.filter(tab =>
				tab.tabId !== action.tabId);
		case TOGGLE_VISIBILITY_FILTER:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						showCompletedTodo: !tab.showCompletedTodo
					} :
					tab
			));
		case STAR_TOGGLE_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						starredTodos: todos(tab.starredTodos, action)
					} : tab
			));
		case ADD_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action)
					} : tab
			));
		case ADD_STARRED_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						starredTodos: todos(tab.starredTodos, action),
						todos: todos(tab.todos, action)
					} : tab
			));
		default:
			return state;
	}
}

function todoInfo(state = {}, action) {
	switch (action.type) {
		case STAR_TOGGLE_TODO:
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
		case ACTIVE_TODO:
			return {
				...state,
				[action.todoId]: todo(state[action.todoId], action)
			};
		default:
			return state;
	}
}

const tabs = combineReducers({
	todoInfo,
	tabInfo
});

export default tabs;

export function getTabs(state) {
	const tabs = state.tabs.tabInfo.map(tab => {
		const todos = tab.todos.map(todo => (
			state.tabs.todoInfo[todo]
		));
		const starredTodos = tab.starredTodos.map(todo => (
			state.tabs.todoInfo[todo]
		));
		return {
			...tab,
			todos,
			starredTodos
		}
	});
	return tabs;
}
