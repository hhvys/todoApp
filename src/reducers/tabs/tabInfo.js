import {
	ADD_TODO,
	ADD_STARRED_TODO,
	STAR_TOGGLE_TODO,
	SORT_BY,
	CHANGE_SORT,
	ADD_TAB,
	INBOX_ID,
	STARRED_ID,
	COPY_TAB,
	CHANGE_TAB_NAME,
	TOGGLE_VISIBILITY_FILTER,
	DELETE_TAB,
	TOGGLE_TODO
} from "../../actions/actionTypes";
import {getTabs} from "./tabs";

/**
 *
 * @param todos - array of to-do ids.
 * @param sortBy
 * @param todoById
 * @returns {*} - sorted array of to-do ids.
 */

const stableSortTodos = (todos, sortBy, todoById) => {
	//For stable sort
	const indexOf = {};
	todos.forEach((todo, index) => {
		indexOf[todo] = index;
	});

	const compareByIndex = (a, b) => {
		if (indexOf[a.todoId] < indexOf[b.todoId])
			return -1;
		return 1;
	};
	return todos.sort((todo1, todo2) => {
		const a = todoById[todo1];
		const b = todoById[todo2];
		switch (sortBy) {
			case SORT_BY.SORT_PRIORITY:
				if (a.star && !b.star)
					return -1;
				if (!a.star && b.star)
					return 1;
				return compareByIndex(a, b);

			case SORT_BY.SORT_CREATION:

				if (a.createdTime > b.createdTime)
					return -1;
				if (a.createdTime < b.createdTime)
					return 1;
				return compareByIndex(a, b);

			case SORT_BY.SORT_ALPHA:
				if (a.text.toLowerCase() < b.text.toLowerCase())
					return -1;
				if (a.text.toLowerCase() > b.text.toLowerCase())
					return 1;
				return compareByIndex(a, b);
		}
	});
};

const todos = (state = [], action, todoById) => {
	switch (action.type) {

		case CHANGE_SORT:
			const newState = [...state];
			return stableSortTodos(newState, action.sortBy, todoById);

		case ADD_TODO:
		case ADD_STARRED_TODO:
			return (
				[
					...state,
					action.todoId
				]);

		case STAR_TOGGLE_TODO:

			/*
				Add to-do id to state array if to-do id is not present and remove if present.
			 */

			const index = state.findIndex(todoId => action.todoId === todoId);
			if (index === -1) {
				return [
					...state,
					action.todoId
				];
			}
			const updatedState = state;
			updatedState.splice(index, 1);
			return updatedState;

		default:
			return state;
	}
};

export const initialState = [
	{
		tabId: INBOX_ID,
		tabName: 'Inbox',
		todos: [],
		starredTodos: [],
		inCompletedTodos: 0
	},
	{
		tabId: STARRED_ID,
		tabName: 'Starred',
		todos: [],
		starredTodos: [],
		inCompletedTodos: 0
	}
];

function tabInfo(state = initialState, action, todoById) {
	let newState;
	switch (action.type) {
		case CHANGE_SORT:
			return state.map(tab => {
				if (tab.tabId !== action.tabId)
					return tab;
				return ({
					...tab,
					todos: todos(tab.todos, action, todoById),
					starredTodos: todos(tab.starredTodos, action, todoById)
				})
			});

		case ADD_TAB:
			return [
				...state,
				{
					tabId: action.tabId,
					tabName: action.tabName,
					todos: [],
					starredTodos: [],
					inCompletedTodos: 0
				}
			];

		case CHANGE_TAB_NAME:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						tabId: tab.tabId,
						tabName: action.tabName,
					} :
					tab
			));

		case COPY_TAB:

			newState = state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = 0;
					Object
						.keys(action.todos)
						.forEach(todo => {
							if (!action.todos[todo].completed && action.todos[todo].star)
								change += 1;
						});
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					}
				}
				return tab;
			});

			return (
				[
					...newState,
					{
						tabId: action.toId,
						tabName: action.tabName,
						todos: Object
							.keys(action.todos)
							.map(key => action.todos[key].todoId),
						starredTodos: [...action.starredTodos],
						inCompletedTodos: action.inCompletedTodos
					}
				]
			);
		case DELETE_TAB:
			newState = state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = 0;
					action
						.todos
						.forEach(todo => {
							if (!todoById[todo].completed && todoById[todo].star)
								change += 1;
						});
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos - change
					}
				}
				return tab;
			});
			return newState.filter(tab =>
				tab.tabId !== action.tabId);
		case TOGGLE_VISIBILITY_FILTER:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						showCompletedTodo: !tab.showCompletedTodo
					} :
					tab
			));
		case STAR_TOGGLE_TODO:
			return state.map(tab => {
				if (tab.tabId === STARRED_ID) {
					let change = action.star ? -1 : +1;
					if (action.completed)
						change = 0;
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					};
				}
				if (tab.tabId === action.tabId) {
					return {
						...tab,
						starredTodos: todos(tab.starredTodos, action),
					};
				}
				return tab;
			});
		case ADD_TODO:
			return state.map(tab => (
				tab.tabId === action.tabId ?
					{
						...tab,
						todos: todos(tab.todos, action),
						inCompletedTodos: tab.inCompletedTodos + 1
					} :
					tab
			));
		case ADD_STARRED_TODO:
			return state.map(tab => {
				if (tab.tabId === action.tabId)
					return {
						...tab,
						starredTodos: todos(tab.starredTodos, action),
						todos: todos(tab.todos, action),
						inCompletedTodos: tab.inCompletedTodos + 1
					};
				if (tab.tabId === STARRED_ID)
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + 1
					};
				return tab
			});
		case TOGGLE_TODO:
			const changeStarredTab = action.star;

			return state.map(tab => {
				if (tab.tabId === action.tabId || (changeStarredTab && tab.tabId === STARRED_ID)) {
					const change = !action.completed ? -1 : +1;
					return {
						...tab,
						inCompletedTodos: tab.inCompletedTodos + change
					};
				}
				return tab;
			});
		default:
			return state;
	}
}

export const getTabInfo = (state, tabId) => {
	state = getTabs(state);
	if (typeof tabId === 'undefined')
		return state.tabInfo;
	return state
		.tabInfo
		.find(tab => tab.tabId === tabId);
};

export default tabInfo;
