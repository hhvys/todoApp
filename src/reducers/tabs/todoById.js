import {
	ADD_STARRED_TODO,
	ADD_TODO,
	COPY_TAB, DELETE_TAB,
	STAR_TOGGLE_TODO,
	TOGGLE_TODO
} from "../../actions/actionTypes";
import {getTabs} from "./tabs";

/**
 *
 * @param state - Object containing information for particular todo
 * @param action
 * @returns {*} - Updated object containing information for todo
 */

const updateTodoInfo = (state = {}, action) => {
	switch (action.type) {
		case ADD_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				createdTime: new Date(),
			};
		case ADD_STARRED_TODO:
			return {
				tabId: action.tabId,
				todoId: action.todoId,
				text: action.text,
				star: true,
				createdTime: new Date(),
			};
		case TOGGLE_TODO:
			return {
				...state,
				completed: !state.completed,
				completedTime: new Date()
			};
		case STAR_TOGGLE_TODO:
			return {
				...state,
				star: !state.star
			};
		default:
			return state;
	}
};

function todoById(state = {}, action) {
	switch (action.type) {
		case STAR_TOGGLE_TODO:
		case ADD_TODO:
		case ADD_STARRED_TODO:
		case TOGGLE_TODO:
			state[action.todoId] = updateTodoInfo(state[action.todoId], action);
			return state;
		case COPY_TAB:
			return {...state, ...(action.todos)};
		case DELETE_TAB:
			const newState = {...state};
			action.todos.forEach(todo => delete newState[todo]);
			return newState;
		default:
			return state;
	}
}

export default todoById;



/**
 * Selector from todoById
 * @param state - full state
 * @param todos - single todoId or array of todoIds
 * @returns {*} - object or array of object with full todoInfo
 */

export function getTodoById(state, todos) {
	state = getTabs(state);
	if (Array.isArray(todos)) {
		return todos.map(todo => (
			state.todoById[todo]
		))
	}
	return state.todoById[todos];
}
