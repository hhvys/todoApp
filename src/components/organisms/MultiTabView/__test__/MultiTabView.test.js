import React from 'react';
import MultiTabView from '../MultiTabView';
import tabs from './tabsInput';

describe('MultiTabView', () => {
	let props;
	let mountedMultiTab;

	const multiTabView = () => {
		if (!mountedMultiTab) {
			mountedMultiTab = shallow(
				<MultiTabView {...props}/>
			);
		}
		return mountedMultiTab;
	};

	beforeEach(() => {
		props = {
			tabs: [],
			onHeaderSymbolClick: undefined,
			onFooterSymbolClick: undefined,
			searchQuery: '',
			onButtonClick: jest.fn(),
			collapsed: undefined,
			onInputSubmit: undefined
		};
		mountedMultiTab = undefined;
	});

	it('should render MultiTabView', () => {
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should render InputWithLabel', () => {
		props.searchQuery = '';
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should render NotFound', () => {
		props.searchQuery = 'searchQuery';
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should have rendered number of Buttons = tabs.length', () => {
		props.tabs = tabs;
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should pass onButtonClick to Button as onClick', () => {
		props.tabs = tabs;
		props.onButtonClick = jest.fn();
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should pass onFooterSymbolClick to RowComponent', () => {
		props.tabs = tabs;
		props.onFooterSymbolClick = jest.fn();
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

	it('should pass onHeaderSymbolClick to RowComponent', () => {
		props.tabs = tabs;
		props.onHeaderSymbolClick = jest.fn();
		expect(toJSON(multiTabView())).toMatchSnapshot();
	});

});
