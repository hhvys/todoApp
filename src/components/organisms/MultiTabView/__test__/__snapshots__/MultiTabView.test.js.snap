// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiTabView should have rendered number of Buttons = tabs.length 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <div
    key="9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="1ST TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="9c60c2e7-291c-4e30-9dfe-9cfbb2295476"
      mainContent="tab 1 1"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="e26ee1de-944d-4b3a-88cc-1464fe53f32b"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="2ND TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="899cda93-66bd-483c-a7aa-3852b900013f"
      mainContent="tab 2 3"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
</div>
`;

exports[`MultiTabView should pass onButtonClick to Button as onClick 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <div
    key="9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="1ST TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="9c60c2e7-291c-4e30-9dfe-9cfbb2295476"
      mainContent="tab 1 1"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="e26ee1de-944d-4b3a-88cc-1464fe53f32b"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="2ND TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="899cda93-66bd-483c-a7aa-3852b900013f"
      mainContent="tab 2 3"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
</div>
`;

exports[`MultiTabView should pass onFooterSymbolClick to RowComponent 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <div
    key="9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="1ST TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="9c60c2e7-291c-4e30-9dfe-9cfbb2295476"
      mainContent="tab 1 1"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="e26ee1de-944d-4b3a-88cc-1464fe53f32b"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="2ND TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="899cda93-66bd-483c-a7aa-3852b900013f"
      mainContent="tab 2 3"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
</div>
`;

exports[`MultiTabView should pass onHeaderSymbolClick to RowComponent 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <div
    key="9ad18f5c-d3b8-46ff-a54e-69bba97f1aa1"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="1ST TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="9c60c2e7-291c-4e30-9dfe-9cfbb2295476"
      mainContent="tab 1 1"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
  <div
    key="e26ee1de-944d-4b3a-88cc-1464fe53f32b"
  >
    <Button
      className="mt-4 mb-2"
      onClick={[Function]}
      text="2ND TAB"
    />
    <RowComponent
      className="todo"
      footerClass="logo"
      footerSymbol="STAR"
      headerClass="logo"
      headerSymbol="CHECK_BOX"
      key="899cda93-66bd-483c-a7aa-3852b900013f"
      mainContent="tab 2 3"
      onFooterSymbolClick={[Function]}
      onHeaderSymbolClick={[Function]}
    />
  </div>
</div>
`;

exports[`MultiTabView should render InputWithLabel 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <NotFound />
</div>
`;

exports[`MultiTabView should render MultiTabView 1`] = `
<div
  className="multiTabView  mr-3"
>
  <InputWithLabel
    className="input"
    label="+"
    onSubmit={[Function]}
    placeholder="Add a starred to-do in Inbox..."
  />
  <NotFound />
</div>
`;

exports[`MultiTabView should render NotFound 1`] = `
<div
  className="multiTabView  mr-3"
>
  <NotFound />
</div>
`;
